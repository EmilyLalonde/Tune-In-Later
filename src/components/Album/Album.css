.Album {
    border-radius: 8px;
    border: 4px solid #ed1228;
    height: 200px;
    margin: 6%;
    margin-top: 12%;
    width: 200px;
    cursor: pointer;
    perspective: 600;
    position: relative;
}

.card-border:hover,
.Album:hover {
   box-shadow:
   0 0 0.1vw  0.1vw #fff7f7,   
   0 0 0.3vw  0.3vw #a20818, 
   0 0   1.5vw  0.2vw #ed1228,
   inset 0 0 1vw  0.2vw #ed1228,
   inset 0 0 0.3vw  0.1vw #a20818,
   inset 0 0 0.1vw  0.1vw #fff7f7;
   background-color: rgba(0, 0, 0, .3)
}

.card {
    height: 100%;
    position: absolute;
    transform-style: preserve-3d;
    transition: all 1s ease-in-out;
    width: 100%;
}

.card:hover {
    transform: rotateY(180deg);
}

.card .coverArt {
    backface-visibility: hidden;
    border-radius: 6px;
    height: 100%;
    position: absolute;
    overflow: hidden;
    width: 100%;
}

.card .card-back {
    backface-visibility: hidden;
    background-color: #002236;
    color: white;
    height: 100%;
    text-align: center;
    transform: rotateY(180deg);
}

.album-name,
p {
    margin: 0;
}

h3 {
    margin: 0;
    font-size: 16px;
}

.favorite {
    background-color: rgba(0,0,0,0);
    border: none;
    background-image: url("https://cdn0.iconfinder.com/data/icons/rafif-flat-vol-4/512/heart-512.png");
    background-size: cover;
    height: 25px;
    margin-top: 1%;
    margin-right: 3%;
    width: 25px;
    float: right;
}



.notFavorite {
    background-color: rgba(0,0,0,0);
    border: none;
    background-image: url("https://cdn1.iconfinder.com/data/icons/ui-22/24/496-512.png");
    background-size: cover;
    height: 25px;
    margin-top: 1%;
    margin-right: 3%;
    width: 25px;
    float: right;
}

.card-border {
    /* background-color: #15b0f6; */
    border: 4px solid #ed1228;
    border-radius: 8px;
    height: 250px;
    margin-bottom: 30px;
    width: 250px;
}

.label {
    color: #36abdd;
    font-size: 20px;
}