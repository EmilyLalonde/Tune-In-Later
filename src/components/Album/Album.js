import "./Album.css";
import React from "react";

const Album = ({ albumData, handleFavorite, isFav }) => {
  return (
    <div className="card-border">
      <button className={isFav ? 'favorite' : 'notFavorite'} onClick={(e) => handleFavorite(e, albumData)}>Save for later</button>
    <article className='Album'>
      <div className="card">
      <div className="card-front">
      <img src={albumData.artworkUrl100 || albumData.artwork_url} alt="album cover art" className="coverArt"/>
      </div>
      <div className="card-back">
      <h2 className="album-name">{albumData.artistName || albumData.artist_name}</h2>
      <p>{(albumData.releaseDate || albumData.release_date).substring(0,4)}</p>
      <p>{albumData.collectionName || albumData.album_name}</p>
      </div>
      </div>
    </article>
    </div>
  );
};

export default Album;
