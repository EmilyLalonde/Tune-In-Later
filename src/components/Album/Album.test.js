import React from 'react';
import { shallow } from 'enzyme'
import Album from './Album';

describe('Album', () => {

  let wrapper;
  beforeEach(() => {
    const mockAlbumData = { 
      "artistId": 4022281, 
      "artistName": "Rick Ross", 
      "artworkUrl100": "https://is2-ssl.mzstatic.com/image/thumb/Music128/v4/a8/fc/39/a8fc39b7-3c90-9c5b-5b58-cc6a460662c4/source/100x100bb.jpg",
      "collectionId": 1440784451, 
      "collectionName": "Teflon Don", 
      "releaseDate": "2010-07-20" };
    const mockHandleFavorite = jest.fn();
    const mockIsFav = true;

    wrapper = shallow(<Album 
      albumData={mockAlbumData}
      handleFavorite={mockHandleFavorite}
      isFav={mockIsFav}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})