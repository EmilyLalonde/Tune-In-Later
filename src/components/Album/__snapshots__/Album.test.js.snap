// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album should match the snapshot 1`] = `
<div
  className="card-border"
>
  <button
    className="favorite"
    onClick={[Function]}
  >
    Save for later
  </button>
  <article
    className="Album"
  >
    <div
      className="card"
    >
      <div
        className="card-front"
      >
        <img
          alt="album cover art"
          className="coverArt"
          src="https://is2-ssl.mzstatic.com/image/thumb/Music128/v4/a8/fc/39/a8fc39b7-3c90-9c5b-5b58-cc6a460662c4/source/100x100bb.jpg"
        />
      </div>
      <div
        className="card-back"
      >
        <h2
          className="album-name"
        >
          Rick Ross
        </h2>
        <p>
          2010
        </p>
        <p>
          Teflon Don
        </p>
      </div>
    </div>
  </article>
</div>
`;
